rpmlb/builder/koji.py

```diff
@@ -0,0 +1,173 @@
+"""Builder interface for the Koji build service."""
+
+import re
+from pathlib import Path
+from typing import Iterator, Optional, Sequence
+
+from ..utils import run_cmd, run_cmd_with_capture
+from ..work import Work
+from .base import BaseBuilder
+
+
+class KojiBuilder(BaseBuilder):
+    """Builder interface for the Koji build service."""
+
+    DEFAULT_TARGET_FORMAT = 'sclo{epel}-{collection}-rh-el{epel}'
+
+    def __init__(self):
+        """Initialize the slots"""
+
+        self.collection = None  #: The collection to build
+        self.epel = None  #: Version of EPEL to build for
+        self.profile = None  #: The koji configuration profile to use
+        #: The format string for construction of the actual template name
+        self.target_format = type(self).DEFAULT_TARGET_FORMAT
+
+    @property
+    def base_command(self) -> Sequence[str]:
```

Comment:
In case of creating sub class, there are 2 type of design for object oriented programming.
Those are "is-a" and "has-a"

Ref:
https://en.wikipedia.org/wiki/Is-a
https://en.wikipedia.org/wiki/Has-a

And Builder class's design that I intended, the relation of `BaseBuilder` and the sub classes is "is-a".

In the case of "is-a". `KojiBuilder` is a `BaseBuilder`.
That means that `KojiBuilder` should not expand a responsibility of `BaseBuilder`.
That means `KojiBuilder` specified functions such as `base_command` should be private method. `_base_command`.